### ISSUE TITLE
False positive with useless-super-delegation for required keyword args

### ISSUE THREAD

**Issue by nicktimko** [PyCQA/pylint#2022]
False positive for useless-super-delegation for *required keyword arguments*. Subtle difference from #2006, a corner-case where #1618 fixing #1085 wasn't quite complete.

### Steps to reproduce
```python
# uselesssuper.py
class MyBase:
    def method(self, *, arg):
        self.x = arg

class MyChild(MyBase):
    def method(self, *, arg="I have value!"):
        super().method(arg=arg)

myobj = MyChild()
myobj.method()
print(myobj.x)
```

### Current behavior
```bash
$ pylint --version
Using config file .../.pylintrc
pylint 1.8.4, 
astroid 1.6.3
Python 3.6.4 (default, Mar 26 2018, 16:21:07) 
[GCC 5.4.0 20160609]

$ python uselesssuper.py
I have value!

$ pylint uselesssuper.py --disable=all --enable=useless-super-delegation
Using config file .../.pylintrc
************* Module uselesssuper
W:  7, 4: Useless super delegation in method 'method' (useless-super-delegation)

------------------------------------------------------------------
Your code has been rated at 8.89/10 (previous run: 1.11/10, +7.78)
```

### Expected behavior

No issue.

**Comment by PCManticore** [comment]
Thanks, makes sense!

**Comment by filip-zyzniewski** [comment]
Hi, it seems to still be the case for default dict values (I know, bad idea in general):
```python
# uselesssuper.py
class MyBase:
    def method(self, *, arg={}):
        self.x = arg

class MyChild(MyBase):
    def method(self, *, arg={'foo': 'bar'}):
        super().method(arg=arg)

myobj = MyChild()
myobj.method()
print(myobj.x)
```

```shell
$ pipenv run pylint --version
pylint 2.5.3
astroid 2.4.2
Python 3.8.5 (default, Jul 21 2020, 10:42:08) 
[Clang 11.0.0 (clang-1100.0.33.17)]
$ python3 uselesssuperdict.py 
{'foo': 'bar'}
$ pipenv run pylint uselesssuperdict.py --disable=all --enable=useless-super-delegation
************* Module uselesssuperdict
uselesssuperdict.py:7:4: W0235: Useless super delegation in method 'method' (useless-super-delegation)

------------------------------------------------------------------
Your code has been rated at 8.89/10 (previous run: 8.89/10, +0.00)

$ 
```
